tests/trainer/test_fsdp.py

```diff
@@ -42,7 +54,8 @@ def test_fsdp_device_initialization(model: ComposerClassifier, mixed_precision:
         train_dataloader=dataloader,
         fsdp_config={
             'activation_checkpointing_reentrant': reentrant,
-            'mixed_precision': mixed_precision
+            'mixed_precision': mixed_precision,
+            'sync_module_states': True if device == 'mixed' else False,
```

Comment:
why is this true for mixed?

Reply:
for mixed init, the weights need to come from rank 0, rather than initialized on each rank. that is what the `sync_module_states` flag does.